src/test/java/core/basesyntax/UserRegistrationTest.java

```diff
@@ -0,0 +1,116 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class UserRegistrationTest {
+    private static final String SUCCESS_LOGIN = "SUCCESS_LOGIN";
+    private static final String SUCCESS_PASSWORD = "SUCCESS_PASSWORD_123";
+    private static final Integer SUCCESS_AGE = 19;
+    private static final String NOT_SUCCESS_LOGIN = "short";
+    private static final String NOT_SUCCESS_PASSWORD = "short";
+    private static final Integer NOT_SUCCESS_AGE = 15;
+    private RegistrationServiceImpl registrationService;
+    private final static User user = new User();
+
+    @BeforeAll
+    static void setUserValidParams() {
+        user.setLogin(SUCCESS_LOGIN);
+        user.setPassword(SUCCESS_PASSWORD);
+        user.setAge(SUCCESS_AGE);
+    }
+
+    @AfterEach
+    void setValidUserParams() {
+        user.setLogin(SUCCESS_LOGIN);
+        user.setPassword(SUCCESS_PASSWORD);
+        user.setAge(SUCCESS_AGE);
+        registrationService.removeByLogin(SUCCESS_LOGIN);
+    }
+
+    @BeforeEach
+    void setStorageDao() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void allUserValuesValid_OK() {
+        assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    void nullUser_NotOK() {
+        assertNull(null);
```

Comment:
The `nullUser_NotOK` test case is incorrect. The assertion `assertNull(null)` does not test the registration service. It should be `assertNull(registrationService.register(null))` to verify that registering a null user returns null.